File: nu_algebra.py
class NU:
    def __init__(self, n, u):
        self.n = n
        self.u = u
    def __repr__(self):
        return f"({self.n}, {self.u})"
    def add(self, other):
        return NU(self.n + other.n, self.u + other.u)
    def mul(self, other):
        return NU(self.n * other.n, self.n * other.u + other.n * self.u)
    def scalar(self, a):
        return NU(a * self.n, abs(a) * self.u)
# Example usage:
x = NU(10, 1)
y = NU(5, 0.5)
print("x + y =", x.add(y))
print("x * y =", x.mul(y))
print("2 * x =", x.scalar(2))

File: src/nu_algebra.R
NU_add <- function(x, y) {
  c(n = x[1] + y[1], u = x[2] + y[2])
}
NU_mul <- function(x, y) {
  c(n = x[1] * y[1], u = x[1] * y[2] + y[1] * x[2])
}
NU_scalar <- function(a, x) {
  c(n = a * x[1], u = abs(a) * x[2])
}
# Example usage:
x <- c(10, 1)   # (n, u)
y <- c(5, 0.5)
cat("x + y =", NU_add(x, y), "\n")
cat("x * y =", NU_mul(x, y), "\n")
cat("2 * x =", NU_scalar(2, x), "\n")
